models/models.js

```diff
@@ -1,8 +1,26 @@
 const db = require("../db/connection")
+const fs = require("fs/promises")
+const path = require("path");
+
 
 
 exports.selectTopics = () => {
     return db.query('SELECT * FROM topics;').then((topics) => {
         return topics.rows
     })
-}
\ No newline at end of file
+}
+
+exports.selectEndpoints = () => {
+    const filePath = path.resolve(__dirname, '../endpoints.json')
+    return fs.readFile(filePath, 'utf-8')
+     .then((data) => {
+        return data
+        
+    }).catch((err) => {
```

Comment:
you don't need .catch() in the model function just in your controller.
If the model function fails as it is invoked within the controller function the catch in the controller will catch the err and pass the on using next